linkerd/app/outbound/src/ingress.rs

```diff
@@ -248,3 +239,7 @@ impl<B> svc::stack::RecognizeRoute<http::Request<B>> for TargetPerRequest {
 #[derive(Debug, Error)]
 #[error("ingress routing requires a service profile")]
 struct ProfileRequired;
+
+#[derive(Debug, Error)]
+#[error("ingress routing requires the l5d-dst-override header")]
+struct DstOverrideRequired;
```

Comment:
could maybe be worth having the error indicate whether the header was missing or malformed? but, that would take a somewhat larger change, so it's not a blocker.